### DISCUSSION TITLE
Some Matplotlib Knowledge

### DISCUSSION THREAD

**Discussion by yuchien-lin** [1101-GS4719-C/Python-Community-GS4719-C#1637] [General]
use  " fig = plt.figure()  ->  plt.plot(x)  " then use use  "fig,ax = plt.subplots()  -> ax.plot(x) 

or use  "fig,ax = plt.subplots()  ->  ax.plot(x)" then use " fig = plt.figure()  ->  plt.plot(x)  " 
it only show the latter picture  

it is 

> fig, ax = plt.subplots()   ax.plot(x) 

![1011-1](https://github.com/NCU-GS4719-Python/Python-Community/assets/71421116/662704c4-323a-46ab-83a8-0ba1b3115ae0)
![1011-2](https://github.com/NCU-GS4719-Python/Python-Community/assets/71421116/0b9a8689-ae3c-4d3b-866c-9a62ecc8c047)

CODE :
> import matplotlib.pyplot as plt
> x = [1,2,3,4,5,6,7,8,9,10]
> y = [66,55,44,88,33,55,99,77,66,44]
> 
> fig = plt.figure(num = 1)
> plt.subplot()
> plt.plot(x,y )
> fig,ax = plt.subplots(num = 1)
> ax.plot(x,y , color = "yellow" ,linestyle = "--")
> 
> plt.show()

it is 

>fig = plt.figure()    plt.plot(x) 

![1011-3](https://github.com/NCU-GS4719-Python/Python-Community/assets/71421116/49f9cbcd-4ed3-4463-a7ae-b1788ca1f3b3)
![1011-4](https://github.com/NCU-GS4719-Python/Python-Community/assets/71421116/eaa06e49-fc06-49e8-9b43-4a75f9df7366)

CODE :
> import matplotlib.pyplot as plt
> x = [1,2,3,4,5,6,7,8,9,10]
> y = [66,55,44,88,33,55,99,77,66,44]
> 
> fig,ax = plt.subplots(num = 1)
> ax.plot(x,y , color = "yellow" ,linestyle = "--")
> fig = plt.figure(num = 1)
> plt.subplot()
> plt.plot(x,y )
> 
> plt.show()
> 

from two different can understand  we can not simultaneously use this two different on the same figure ,otherwise there will be many figure can not show . Consequently , we need to decide which way do you want to use .No matter which , it also have its advantage ,so we  need learn more about it . Then we use it more convenient.

**Comment by 1chooo** [comment]
程式碼可以用 ``` 的方式框住，如此不只可以有等寬字體，也可以加入個程式語言的高亮

\`\`\`python
print("Hello GS4719")
\`\`\`

**效果如下：**
```python
print("Hello GS4719")
```

如果想在一段文字描述也加上程式碼的敘述，像是 word = "Hello World" ，可以用 \`\` 用一個包起來

**效果如下：**

如果想在一段文字描述也加上程式碼的敘述，像是 `word = "Hello World"`

幫你把文中的程式碼加上了！
---
```py
import matplotlib.pyplot as plt
x = [1,2,3,4,5,6,7,8,9,10]
y = [66,55,44,88,33,55,99,77,66,44]

fig = plt.figure(num = 1)
plt.subplot()
plt.plot(x,y )
fig,ax = plt.subplots(num = 1)
ax.plot(x,y , color = "yellow" ,linestyle = "--")

plt.show()
```

```py
fig = plt.figure() plt.plot(x)
```

```py
import matplotlib.pyplot as plt
x = [1,2,3,4,5,6,7,8,9,10]
y = [66,55,44,88,33,55,99,77,66,44]

fig,ax = plt.subplots(num = 1)
ax.plot(x,y , color = "yellow" ,linestyle = "--")
fig = plt.figure(num = 1)
plt.subplot()
plt.plot(x,y )

plt.show()
```

最後感謝同學的整理！
